index.html

```diff
@@ -101,19 +199,623 @@ <h3>
+
+          <pre class="js">
+            socket.closed.then(() => console.log("Closed"), () => console.log("Errored"));
+          </pre>
+
+          Note that {{TCPSocket/closed}} will be automatically rejected or resolved once both
+          {{TCPSocket/[[readable]]}} and {{TCPSocket/[[writable]]}} reach an unreadable state.
+        </aside>
+       
+        When called, returns the [=this=].{{TCPSocket/[[closedPromise]]}}.
+      </section>
+
+      <section>
+        <h3><dfn>close()</dfn> method</h3>
+
+        <aside class="example">
+          When communication with the port is no longer required it can be closed
+          and the associated resources released by the system.
+
+          <p>
+          Calling `socket.`{{UDPSocket/close()}} implicitly invokes
+          `connection.`{{UDPSocketConnection/readable}}`.`{{ReadableStream/cancel()}} and
+          `connection.`{{UDPSocketConnection/writable}}`.`{{WritableStream/abort()}} in order to
+          clear any buffered data. If the application has called
+          `connection.`{{UDPSocketConnection/readable}}`.`{{ReadableStream/getReader()}} or
+          `connection.`{{UDPSocketConnection/writable}}`.`{{WritableStream/getWriter()}} the stream
+          is locked and the socket cannot be closed. This forces the developer to
+          decide how to handle any read or write operations that are in progress.
+
+
+          For example, to ensure that all buffered data has been transmitted before
+          the socket is closed the application must await the {{Promise}} returned by
+          `writer.`{{WritableStreamDefaultWriter/close()}}.
+
+          <pre class="js">
+            const encoder = new TextEncoder();
+            const { writable } = await socket.connection;
+            const writer = writable.getWriter();
+
+            writer.write(encoder.encode("A long message that will take..."));
+            await writer.close();
+            await socket.close();
+          </pre>
+
+          To discard any unsent data the application could instead call
+          `writer.`{{WritableStreamDefaultWriter/abort()}}.
+
+          If a loop is being used to read data from the socket, then it must be exited before calling `socket`.{{TCPSocket/close()}}.
+
+          <pre class="js">
+            const { readable } = await socket.connection;
+            const reader = readable.getReader();
+
+            async function readUntilClosed() {
+              while (true) {
+                const { value, done } = await reader.read();
+                if (done) {
+                  // |reader| has been canceled.
+                  break;
+                }
+                // Do something with |value|...
+              }
+
+              await socket.close();
+            }
+
+            const read_complete = readUntilClosed();
+
+            // Sometime later...
+            reader.cancel();
```

Comment:
The `ReadableStreamDefaultReader` interface has been updated so now you can call `releaseLock()` instead of `cancel()` to cancel a call to `read()` without invoking the cancel algorithm, which is better if the intention is to break out of the loop without actually closing the socket. This is available in M-105.

Reply:
Fixed!